docs/src/main/sphinx/connector/faker.md

```diff
@@ -0,0 +1,225 @@
+# Faker connector
+
+The Faker connector generates random data matching a defined structure. It uses
+the [Datafaker](https://www.datafaker.net/) library, which makes the generated
+data more realistic. It can be used to easily populate another data source with
+large and realistic test data, which can help with testing performance of
+applications processing data (like Trino itself), and also application's user
+interface.
+
+## Usage
+
+Faker generates data when reading from a table created in a catalog using this
+connector. This makes it easy to fill an existing schema with random data, by
+copying only the schema into a Faker catalog, and inserting the data back into
+the original tables.
+
+Create a catalog properties file that specifies the Faker connector by setting
+the `connector.name` to `faker`.
+
+For example, to access data as the `generator` catalog, create the file
+`etc/catalog/generator.properties`. Replace the connection properties as
+appropriate for your setup:
+
+
+```text
+connector.name=faker
+faker.null-probability=0.1
+faker.default-limit=1000
+```
+
+Create a table with the same columns as in the table to be populated with
+random data. Exclude all properties, because the Faker connector doesn't
+support same properties as other connectors.
+
+```sql
+CREATE TABLE generator.default.customer (LIKE production.public.customer EXCLUDING PROPERTIES);
+```
+
+Insert random data into the original table, by selecting it from the Faker
+connector. Data generated for columns of non-character types cover the whole
+range of that data type, so to make it more practical, adding constraints might
```

Comment:
```suggestion
range of that data type. Add constraints to adjust the data as desired. For example the following example ensures that date of birth and age in years are related and realistic values.
```

also .. maybe adjust to <100 or even <80

Reply:
Be optimistic! Jimmy Carter recently had his 100 birthday, and can't play with Lego anymore :-P